File: latexdraw-core/net.sf.latexdraw/src/main/net/sf/latexdraw/actions/UpdateTemplates.scala
package net.sf.latexdraw.actions

import org.malai.action.Action

import net.sf.latexdraw.generators.svg.SVGDocumentGenerator

/**
 * This action updates the templates.<br>
 * <br>
 * This file is part of LaTeXDraw.<br>
 * Copyright (c) 2005-2013 Arnaud BLOUIN<br>
 * <br>
 * LaTeXDraw is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation; either version 2 of the License, or (at your option) any later version.
 * <br>
 * LaTeXDraw is distributed without any warranty; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * General Public License for more details.<br>
 * <br>
 * 2012-04-19<br>
 * @author Arnaud BLOUIN
 * @since 3.0
 */
class UpdateTemplates extends Action with TemplateAction{
	/** Defines if the thumbnails must be updated. */
	protected var _updateThumbnails : Boolean = false


	override def isRegisterable() = false

	override def doActionBody() {
		SVGDocumentGenerator.INSTANCE.updateTemplates(_templatesMenu.get, _updateThumbnails)
	}

	override def canDo() = _templatesMenu.isDefined

	/**
	 * @param updateThumbnails Defines if the thumbnails must be updated.
	 */
	def updateThumbnails_=(update : Boolean) { _updateThumbnails = update }

	def updateThumbnails = _updateThumbnails
}

File: latexdraw-core/net.sf.latexdraw/src/main/java/net/sf/latexdraw/actions/UpdateTemplates.java
/*
 * This file is part of LaTeXDraw.
 * Copyright (c) 2005-2018 Arnaud BLOUIN
 * LaTeXDraw is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation; either version 2 of the License, or (at your option) any later version.
 * LaTeXDraw is distributed without any warranty; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * General Public License for more details.
 */
package net.sf.latexdraw.actions;

import javafx.scene.layout.Pane;
import net.sf.latexdraw.view.svg.SVGDocumentGenerator;
import org.malai.action.ActionImpl;

/**
 * This action updates the templates.
 * @author Arnaud Blouin
 */
public class UpdateTemplates extends ActionImpl implements TemplateAction {
	/** Defines if the thumbnails must be updated. */
	private boolean updateThumbnails;
	private Pane templatesPane;

	public UpdateTemplates() {
		super();
		updateThumbnails = false;
	}

	@Override
	public void doActionBody() {
		SVGDocumentGenerator.INSTANCE.updateTemplates(templatesPane, updateThumbnails);
	}

	/**
	 * @param update Defines if the thumbnails must be updated.
	 */
	public void updateThumbnails(final boolean update) {
		updateThumbnails = update;
	}

	@Override
	public void setTemplatesPane(final Pane templatePanel) {
		templatesPane = templatePanel;
	}

	@Override
	public Pane getTemplatesPane() {
		return templatesPane;
	}
}
